samples: sample.sensor.accel_trig.shield-tap: build failure on several NXP platfrom

**Describe the bug**


sample.sensor.accel_trig.shield-tap below failure on below platfrom:

|Platform|
|lpcxpresso55s16/lpc55s16|
|mimxrt1010_evk/mimxrt1011|
|lpcxpresso54114/lpc54114/m4 |
|rddrone_fmuk66/mk66f18|

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - list above
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

```
f63bb9cafa93bee8c29ccbf9645fe63b0c1e6268 is the first bad commit
commit f63bb9cafa93bee8c29ccbf9645fe63b0c1e6268
Author: Pierrick Curt <pierrickcurt@gmail.com>
Date:   Tue Feb 18 13:58:59 2025 +0100

    samples: sensor: accel_trig: fix twister tests
    
    Add twister tests for tap detection feature
    
    The overlay must be applied only if the shield x_nucleo_iks01a3
    is enabled on board nrf52dk/nrf52832.
    
    Signed-off-by: Pierrick Curt <pierrickcurt@gmail.com>

 samples/sensor/accel_trig/README.rst               |  1 +
 .../accel_trig/boards/nrf52dk_nrf52832.overlay     | 20 ---------------
 samples/sensor/accel_trig/sample.yaml              | 30 ++++++++++++++++++++++
 samples/sensor/accel_trig/x_nucleo_iks01a3.overlay | 20 +++++++++++++++
 4 files changed, 51 insertions(+), 20 deletions(-)
 delete mode 100644 samples/sensor/accel_trig/boards/nrf52dk_nrf52832.overlay
 create mode 100644 samples/sensor/accel_trig/x_nucleo_iks01a3.overlay
```

**To Reproduce**

```
west twister -p rddrone_fmuk66/mk66f18 -s samples/sensor/accel_trig/sample.sensor.accel_trig.generic-tap
```

**Expected behavior**

Test PASS

**Impact**

CI

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/samples/sensor/accel_trig/src/main.c
In file included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/gcc.h:98,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain.h:52,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/kernel_includes.h:23,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/kernel.h:17,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/samples/sensor/accel_trig/src/main.c:8:
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/samples/sensor/accel_trig/src/main.c: In function 'main':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_N_ALIAS_accel0_ORD' undeclared (first use in this function)
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/common.h:137:26: note: in definition of macro '_DO_CONCAT'
  137 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:96:33: note: in expansion of macro '_CONCAT'
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:263:37: note: in expansion of macro 'DEVICE_NAME_GET'
  263 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:280:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  280 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/samples/sensor/accel_trig/src/main.c:46:42: note: in expansion of macro 'DEVICE_DT_GET'
   46 |         const struct device *const dev = DEVICE_DT_GET(DT_ALIAS(accel0));
      |                                          ^~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:96:41: note: each undeclared identifier is reported only once for each function it appears in
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/common.h:137:26: note: in definition of macro '_DO_CONCAT'
  137 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:96:33: note: in expansion of macro '_CONCAT'
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:263:37: note: in expansion of macro 'DEVICE_NAME_GET'
  263 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:280:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  280 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/samples/sensor/accel_trig/src/main.c:46:42: note: in expansion of macro 'DEVICE_DT_GET'
   46 |         const struct device *const dev = DEVICE_DT_GET(DT_ALIAS(accel0));
      |                                          ^~~~~~~~~~~~~
ninja: build stopped: subcommand failed.

```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v4.1.0-rc2-177-g5efaf9d8c1f0

--- a/boards/nxp/rddrone_fmuk66/rddrone_fmuk66.dts
+++ b/boards/nxp/rddrone_fmuk66/rddrone_fmuk66.dts
@@ -25,6 +25,7 @@
 		red-pwm-led = &red_pwm_led;
 		green-pwm-led = &green_pwm_led;
 		blue-pwm-led = &blue_pwm_led;
+		accel0 = &fxos8700;
 	};
 
 	chosen {
@@ -347,7 +348,7 @@ zephyr_udc0: &usbotg {
 	clock-frequency = <DT_FREQ_M(1)>;
 
 
-	fxos8700@0 {
+	fxos8700: fxos8700@0 {
 		compatible = "nxp,fxos8700";
 		reg = <0>;
 		spi-max-frequency = <DT_FREQ_M(1)>;
@@ -357,7 +358,7 @@ zephyr_udc0: &usbotg {
 		status = "okay";
 	};
 
-	fxas21002@1 {
+	fxas21002: fxas21002@1 {
 		compatible = "nxp,fxas21002";
 		reg = <1>;
 		spi-max-frequency = <DT_FREQ_M(1)>;

